samcli/commands/init/init_templates.py

```diff
@@ -66,7 +67,12 @@ def clone_templates_repo(self):
         if not self._git_repo.clone_attempted:
             shared_dir: Path = GlobalConfig().config_dir
             try:
-                self._git_repo.clone(clone_dir=shared_dir, clone_name=APP_TEMPLATES_REPO_NAME, replace_existing=True)
+                self._git_repo.clone(
```

Comment:
Can we have two step process of clone and checkout instead of combining them into clone?
This is easier to read and we can reuse checkout in the future.